### PR TITLE
[TIC 95] Backend for Manual Refunds

### PR DESCRIPTION
**Pull request by benfaustin95**
### Description
Added backend route to fetch orders associated with an email. 
Added state/functionality to frontend to get orders based on user email. 
Added a confirmation pop-up before refund is processed. 

### Risks
Possibly overlooked edge cases when testing. 

### Validation
Manual testing (place one to many orders -> refund orders -> check database)

### Issue
N/A

### Operating System
Apple Silicon

### CODE REVIEW

File: client/src/components/Ticketing/ticketingmanager/Refund/RefundOrderTicket.tsx [hunk 1]
```diff
@@ -27,128 +47,187 @@ const Refund = () => {
...
-        ...show,
-        showPopUp: true,
-        message: `Order ${orderID} successfully refunded`,
-        title: `Order Refund Successful`,
-        success: true,
-      });
+      setOrders((orders) => orders.filter((order) => order.orderid !== orderID));
+      setPopUp(
+        'Refund Successful',
+        `Order ${orderID} successful refunded`,
```
Comment (henrykaus) [inline client/src/components/Ticketing/ticketingmanager/Refund/RefundOrderTicket.tsx:56]: Nitpick, but probably should be `successfully refunded`

File: client/src/components/Ticketing/ticketingmanager/Refund/RefundOrderTicket.tsx [hunk 2]
```diff
@@ -27,128 +47,187 @@ const Refund = () => {
...
+      if (!response.ok) {
+        throw response;
+      }
+      const data = await response.json();
+      setOrders(data);
+    } catch (error) {
+      console.error(error);
+      setPopUp(
+        'Error',
+        'Return failed',
```
Comment (henrykaus) [inline client/src/components/Ticketing/ticketingmanager/Refund/RefundOrderTicket.tsx:109]: What does `Return failed` mean?

File: client/src/components/Ticketing/ticketingmanager/Refund/RefundOrderTicket.tsx [hunk 3]
```diff
@@ -27,128 +47,187 @@ const Refund = () => {
...
+                    </p>
+                  </td>
                 </tr>
               ) : (
-                mappedInstances.map((instance, index) => (
-                  <tr key={index} className='grid grid-cols-5 gap-2 bg-gray-200 rounded-lg shadow-md px-2 mb-2 hover:bg-gray-300'>
-                    <td className='row-start-1 justify-self-start pl-2 py-2 col-span-1'>
+                orders.map((instance, index) => (
+                  <tr key={index} className = 'grid grid-cols-5 gap-2 bg-gray-200 rounded-lg shadow-md px-2 mb-2 hover:bg-gray-300'>
+                    <td className = 'row-start-1 justify-self-start pl-2 py-2 col-span-1'>
```
Comment (henrykaus) [inline client/src/components/Ticketing/ticketingmanager/Refund/RefundOrderTicket.tsx:181]: Minor, but there shouldn't be spaces around the `=`

File: server/src/controllers/orderController.ts [hunk 4]
```diff
@@ -143,46 +144,95 @@ orderController.use(checkScopes);
...
+        },
+      },
+    });
+    const toReturn = orders.map((order) => {
+      const {
+        contacts,
+        orderitems,
+        ordertotal,
+        orderdate,
+        ...evenythingElse
```
Comment (henrykaus) [inline server/src/controllers/orderController.ts:204]: This looks like a typo

Comment (henrykaus) [reply server/src/controllers/orderController.ts:204]: Also, probably want to use a different name

File: server/src/controllers/orderController.ts [hunk 5]
```diff
@@ -143,46 +144,95 @@ orderController.use(checkScopes);
...
+            }),
+      ).flat()
+          .filter((name, index, array) => name && array.indexOf(name)=== index);
+
+      return {
+        price: ordertotal,
+        email: contacts.email,
+        name: `${contacts.firstname} ${contacts.lastname}`,
+        orderdate: `${orderdate.toString().slice(4, 6)}/${orderdate.toString().slice(6, 8)}/${orderdate.toString().slice(0, 4)}`,
+        ...evenythingElse,
```
Comment (henrykaus) [inline server/src/controllers/orderController.ts:221]: here too

**Review by henrykaus — COMMENTED** [review]
Minor changes, this feature works amazingly!

File: client/src/components/Ticketing/ticketingmanager/Refund/RefundOrderTicket.tsx (hunk 1 above)
Comment (benfaustin95) [reply client/src/components/Ticketing/ticketingmanager/Refund/RefundOrderTicket.tsx]: FIxed

File: client/src/components/Ticketing/ticketingmanager/Refund/RefundOrderTicket.tsx (hunk 2 above)
Comment (benfaustin95) [reply client/src/components/Ticketing/ticketingmanager/Refund/RefundOrderTicket.tsx]: fixed

File: client/src/components/Ticketing/ticketingmanager/Refund/RefundOrderTicket.tsx (hunk 3 above)
Comment (benfaustin95) [reply client/src/components/Ticketing/ticketingmanager/Refund/RefundOrderTicket.tsx]: fixed

File: server/src/controllers/orderController.ts (hunk 4 above)
Comment (benfaustin95) [reply server/src/controllers/orderController.ts]: fixed

File: server/src/controllers/orderController.ts (hunk 5 above)
Comment (benfaustin95) [reply server/src/controllers/orderController.ts]: fixed

**Review by henrykaus — APPROVED** [review]
Thanks for the quick fixes!!

**Review by jonathantnguyen — APPROVED** [review]
Looks good, no apparent issues during manual testing.

**Pull request merged**